WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dynamiccard/DynamicCardsBuilder.kt

```diff
@@ -0,0 +1,73 @@
+package org.wordpress.android.ui.mysite.cards.dynamiccard
+
+import org.wordpress.android.fluxc.model.dashboard.CardModel
+import org.wordpress.android.fluxc.model.dashboard.CardModel.DynamicCardsModel.CardOrder
+import org.wordpress.android.ui.deeplinks.handlers.DeepLinkHandlers
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.Dynamic
+import org.wordpress.android.ui.mysite.MySiteCardAndItemBuilderParams.DynamicCardsBuilderParams
+import org.wordpress.android.ui.utils.ListItemInteraction.Companion.create
+import org.wordpress.android.util.UriWrapper
+import org.wordpress.android.util.UrlUtilsWrapper
+import javax.inject.Inject
+
+class DynamicCardsBuilder @Inject constructor(
+    private val urlUtils: UrlUtilsWrapper,
+    private val deepLinkHandlers: DeepLinkHandlers
+) {
+    fun build(params: DynamicCardsBuilderParams, order: CardOrder): List<Dynamic>? {
+        if (!shouldBuildCard(params, order)) {
+            return null
+        }
+        return convertToDynamicCards(params, order)
+    }
+
+    private fun shouldBuildCard(params: DynamicCardsBuilderParams, order: CardOrder): Boolean {
+        if (params.dynamicCards == null || params.dynamicCards.pages.none { it.order == order }) return false
+        return true
+    }
+
+    private fun convertToDynamicCards(params: DynamicCardsBuilderParams, order: CardOrder): List<Dynamic> {
+        val cards = params.dynamicCards?.pages?.filter { it.order == order } ?: emptyList()
```

Comment:
NIT: Just as an alternative `.orEmpty()` can be used here:
```
params.dynamicCards?.dynamicCards?.filter { it.order == order }.orEmpty()
```
No need to change it, btw :)

Reply:
Great suggestion Artyom 👍 I'll use that in the follow up PR that touches the same code https://github.com/wordpress-mobile/WordPress-Android/pull/19801